guardian/transcription-service
# Use signed urls for transcription output

## What does this change?


This PR does the following:
- CDK - Creates a bucket for storing the transcripts output (srt, json & text)
- CDK - gives `s3:PutObject` permission on the bucket to the api lambda
- CDK - gives `s3:GetObject` permission on the bucket to the outputHandler lambda
- API - Creates S3 upload signed urls for the transcript output before sending the message to SQS
- API - Adds the signed urls to the SQS message
- Worker - Uploads the transcripts to the relevant S3 bucket using the signed urls
- Worker - Adds the S3 keys of the uploaded output files to the SNS message
- Output Handler - Downloads the transcripts from the output bucket using the keys provided by the SQS message
- Output Handler - Reads the transcripts files and adds their text to the object that goes to dynamo db

## How to test


Tested in code by doing the following:
- made an api call to `api/send-message` endpoint which is using hardcoded values for the message
- Waited for the instance to scale up
- Checked the logs in the instance and make sure it's all working fine 
- Waited for the email notification
- Checked dynamo db to make sure the record was added

## TODO:
For Prod release we need to add the new config parameter to parameter store
